homeassistant/components/deconz/logbook.py

```diff
@@ -0,0 +1,33 @@
+"""Describe deCONZ logbook events."""
+
+from typing import Callable, Optional
+
+from homeassistant.const import ATTR_DEVICE_ID, CONF_EVENT
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.event import Event
+
+from .const import DOMAIN as DECONZ_DOMAIN
+from .deconz_event import CONF_DECONZ_EVENT, DeconzEvent
+from .device_trigger import _get_deconz_event_from_device_id
+
+
+@callback
+def async_describe_events(
+    hass: HomeAssistant,
+    async_describe_event: Callable[[str, str, Callable[[Event], dict]], None],
+) -> None:
+    """Describe logbook events."""
+
+    @callback
+    def async_describe_deconz_event(event: Event) -> dict:
+        """Describe deCONZ logbook event."""
+        deconz_event: Optional[DeconzEvent] = _get_deconz_event_from_device_id(
```

Comment:
```
def _get_deconz_event_from_device_id(hass, device_id):
    """Resolve deconz event from device id."""
    for gateway in hass.data.get(DOMAIN, {}).values():

        for deconz_event in gateway.events:

            if device_id == deconz_event.device_id:
                return deconz_event

    return None
```

Since we have to iterate all the devices this has at best `O(1)` and at worst `O(n)` performance.  If we have 100s of events and 100s of devices this could slow down logbook.  It would be nice if `_get_deconz_event_from_device_id` had an index/`dict` to quickly lookup by device id.

Reply:
I don't think it's that big of an issue since these are manually operated devices only, remote controls of different types. So events won't be that many, i'd say 20-40 devices maximum and for 99% only one instance of deconz as well.

We can talk about how I'd best do that index and I can create a new PR that solves the issue since I'm eager to understand different aspects of performance improvements :)